File: greenweather/lib/model/transactionModel.dart
class Transactionmodel {
  final int id;
  final String userId;
  final int points;
  final String type;
  final String reason;
  final String createdAt;
  
  Transactionmodel({
    required this.id,
    required this.userId,
    required this.points,
    required this.type,
    required this.reason,
    required this.createdAt,
  });

  factory Transactionmodel.fromJson(Map<String, dynamic> json) {
    return Transactionmodel(
      id: json['id'] as int,
      userId: json['userId'] as String,
      points: json['points'] as int,
      type: json['type'] as String,
      reason: json['reason'] as String,
      createdAt: json['createdAt'] as String,
    );
  }
  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'points': points,
      'type': type,
      'reason': reason,
      'createdAt': createdAt,
    };
  }
}

File: server/src/model/TransactionModel.ts
import { PrismaClient } from "@prisma/client";
import type { Userinputid } from "../types/userType";
import type { Itranscation } from "../types/transcationType";
const prisma = new PrismaClient();

class Transcationmodel {
  async getTransactionbyuserid(data: Userinputid): Promise<Itranscation[]> {
    return await prisma.pointTransaction.findMany({
      where: { userId: data.userId },
      orderBy: {
        createdAt: 'desc',
      },
      select: {
        id: true,
        userId: true,
        points: true,
        type: true,
        createdAt: true,
        reason: true,
      },
    });
  }

  //create transaction
  async createTransaction(data: Itranscation): Promise<any> {
    return await prisma.pointTransaction.create({
      data: {
        userId: data.userId,
        points: data.points,
        type: data.type,
        reason: data.reason,
      },
    });
  }
}
export const transactionModel = new Transcationmodel();
